// turath-sdk/dist/main.js
import $5oumB$wretch from "wretch";
import $5oumB$wretchaddonsqueryString from "wretch/addons/queryString";


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $96cb7a03b537cb37$exports = {};

$parcel$export($96cb7a03b537cb37$exports, "getAuthor", () => $96cb7a03b537cb37$export$31577d79be25397a);
$parcel$export($96cb7a03b537cb37$exports, "getBookFile", () => $96cb7a03b537cb37$export$6512c9928e63f8f4);
$parcel$export($96cb7a03b537cb37$exports, "getBookInfo", () => $96cb7a03b537cb37$export$72daedf8fb838a67);
$parcel$export($96cb7a03b537cb37$exports, "getPage", () => $96cb7a03b537cb37$export$c2a6433281518c91);
$parcel$export($96cb7a03b537cb37$exports, "search", () => $96cb7a03b537cb37$export$d76128d007d19019);
var $e7af378ba90c6fa6$export$874c480dbba767e5;
(function(Includes) {
    Includes["Indexes"] = "indexes";
})($e7af378ba90c6fa6$export$874c480dbba767e5 || ($e7af378ba90c6fa6$export$874c480dbba767e5 = {}));




const $932b4b3755196b46$var$baseApi = (0, $5oumB$wretch)("https://api.turath.io").addon((0, $5oumB$wretchaddonsqueryString));
const $932b4b3755196b46$var$fileApi = (0, $5oumB$wretch)("https://files.turath.io/books");
const $932b4b3755196b46$export$31935d0a766c641b = async (path)=>{
    return $932b4b3755196b46$var$fileApi.url(path).get().json();
};
const $932b4b3755196b46$export$a75c676177490cac = async (path, queryParams = {})=>{
    return $932b4b3755196b46$var$baseApi.url(path).query(queryParams).get().json();
};


const $96cb7a03b537cb37$var$API_VERSION_NUMBER = 3;
const $96cb7a03b537cb37$export$31577d79be25397a = async (id)=>{
    const queryParams = {
        id: id,
        ver: $96cb7a03b537cb37$var$API_VERSION_NUMBER
    };
    const { info: info } = await (0, $932b4b3755196b46$export$a75c676177490cac)("/author", queryParams);
    if (!info) throw new Error(`Author ${id} not found`);
    return {
        info: info
    };
};
const $96cb7a03b537cb37$export$6512c9928e63f8f4 = async (id)=>{
    try {
        const file = await (0, $932b4b3755196b46$export$31935d0a766c641b)(`/${id}.json`);
        return file;
    } catch (err) {
        if (err.status === 404) throw new Error(`Book ${id} not found`);
        throw err;
    }
};
const $96cb7a03b537cb37$export$72daedf8fb838a67 = async (id)=>{
    const queryParams = {
        id: id,
        include: (0, $e7af378ba90c6fa6$export$874c480dbba767e5).Indexes,
        ver: $96cb7a03b537cb37$var$API_VERSION_NUMBER
    };
    const book = await (0, $932b4b3755196b46$export$a75c676177490cac)(`/book`, queryParams);
    return book;
};
const $96cb7a03b537cb37$export$c2a6433281518c91 = async (bookId, pageNumber)=>{
    const queryParams = {
        book_id: bookId,
        pg: pageNumber,
        ver: $96cb7a03b537cb37$var$API_VERSION_NUMBER
    };
    const { meta: meta, text: text } = await (0, $932b4b3755196b46$export$a75c676177490cac)("/page", queryParams);
    if (!meta && !text) throw new Error(`Book ${bookId}, page ${pageNumber} not found`);
    return {
        meta: JSON.parse(meta),
        text: text
    };
};
const $96cb7a03b537cb37$export$d76128d007d19019 = async (query, { category: category, sortField: sortField, ...options } = {})=>{
    const queryParams = {
        ...options,
        ...category && {
            cat_id: category
        },
        ...sortField && {
            sort: sortField
        },
        q: query,
        ver: $96cb7a03b537cb37$var$API_VERSION_NUMBER
    };
    const { count: count, data: data } = await (0, $932b4b3755196b46$export$a75c676177490cac)("/search", queryParams);
    return {
        count: count,
        data: data.map((r)=>({
                ...r,
                meta: JSON.parse(r.meta)
            }))
    };
};


var $b040cb64458fd080$exports = {};


var $dcb9ae70211e2b58$exports = {};
// Meta information about a book


var $3f2f73f0afb79657$exports = {};

$parcel$export($3f2f73f0afb79657$exports, "SortField", () => $3f2f73f0afb79657$export$c4fb72ec5ef10400);
var $3f2f73f0afb79657$export$c4fb72ec5ef10400;
(function(SortField) {
    SortField["PageId"] = "page_id";
})($3f2f73f0afb79657$export$c4fb72ec5ef10400 || ($3f2f73f0afb79657$export$c4fb72ec5ef10400 = {}));




export {$96cb7a03b537cb37$export$31577d79be25397a as getAuthor, $96cb7a03b537cb37$export$6512c9928e63f8f4 as getBookFile, $96cb7a03b537cb37$export$72daedf8fb838a67 as getBookInfo, $96cb7a03b537cb37$export$c2a6433281518c91 as getPage, $96cb7a03b537cb37$export$d76128d007d19019 as search, $3f2f73f0afb79657$export$c4fb72ec5ef10400 as SortField};
